src/v/storage/log_manager.cc

```diff
@@ -435,8 +435,12 @@ ss::future<> log_manager::housekeeping_loop() {
 
         try {
             co_await housekeeping_scan(lowest_ts_to_retain());
-        } catch (const std::exception& e) {
-            vlog(stlog.info, "Error processing housekeeping(): {}", e);
+        } catch (...) {
+            auto eptr = std::current_exception();
+            if (ssx::is_shutdown_exception(eptr)) {
+                std::rethrow_exception(eptr);
+            }
+            vlog(stlog.warn, "Error processing housekeeping(): {}", eptr);
         }
 
         co_await ss::coroutine::switch_to(prev_sg);
```

Comment:
I'm not seeing how the previous scheduling group is restored when a shutdown exception is rethrown above?

Reply:
> I was under the impression that `ss::coroutine::switch_to` only affected the currently running function, and not the potential chain of function calls in the current fiber, [which would mean] there's no need to switch back in the throw case.

Reply:
yes i read that, but where did you get that impression?

Reply:
From reading [the source for switch_to](https://github.com/scylladb/seastar/blob/master/include/seastar/coroutine/switch_to.hh) and reasoning that the `task` type obtained from a `std::coroutine_handle<T>::promise()` refers to the current co-routine being ran and not the fiber as a whole- but I could be mistaken.

[I also found a historical slack thread that led others to the same conclusion](https://redpandadata.slack.com/archives/C044RD18NMV/p1683225990528099), though I didn't want to accept this as decisive proof on its own.

[Wanted to prove it to myself with an example](https://gist.github.com/WillemKauf/83fff547556a5d7ff83f1867025450ed) that does not switch back before returning from a nested co-routine call, and it seems my impression is correct.

Reply:
cool TIL!